x/auth/middleware/tx_priority.go

```diff
@@ -0,0 +1,62 @@
+package middleware
+
+import (
+	"context"
+
+	abci "github.com/tendermint/tendermint/abci/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/types/tx"
+)
+
+var _ tx.Handler = txPriorityHandler{}
+
+type txPriorityHandler struct {
+	next tx.Handler
+}
+
+// TxPriorityHandler implements tx handling middleware that determines a
+// transaction's priority via a naive mechanism -- the total sum of fees provided.
+// It sets the Priority in ResponseCheckTx only.
+func TxPriorityHandler(txh tx.Handler) tx.Handler {
+	return txPriorityHandler{next: txh}
+}
+
+// CheckTx implements tx.Handler.CheckTx. We set the Priority of the transaction
+// to be ordered in the Tendermint mempool based naively on the total sum of all
+// fees included. Applications that need more sophisticated mempool ordering
+// should look to implement their own fee handling middleware instead of using
+// TxPriorityHandler.
+func (h txPriorityHandler) CheckTx(ctx context.Context, tx sdk.Tx, req abci.RequestCheckTx) (abci.ResponseCheckTx, error) {
+	feeTx, ok := tx.(sdk.FeeTx)
+	if !ok {
+		return abci.ResponseCheckTx{}, sdkerrors.Wrap(sdkerrors.ErrTxDecode, "Tx must be a FeeTx")
+	}
+
+	feeCoins := feeTx.GetFee()
+
+	res, err := h.next.CheckTx(ctx, tx, req)
+	res.Priority = GetTxPriority(feeCoins)
+
+	return res, err
+}
+
+func (h txPriorityHandler) DeliverTx(ctx context.Context, tx sdk.Tx, req abci.RequestDeliverTx) (abci.ResponseDeliverTx, error) {
+	return h.next.DeliverTx(ctx, tx, req)
+}
+
+func (h txPriorityHandler) SimulateTx(ctx context.Context, sdkTx sdk.Tx, req tx.RequestSimulateTx) (tx.ResponseSimulateTx, error) {
+	return h.next.SimulateTx(ctx, sdkTx, req)
+}
+
+// GetTxPriority returns a naive tx priority based on the total sum of all fees
+// provided in a transaction.
+func GetTxPriority(fee sdk.Coins) int64 {
+	var priority int64
+	for _, c := range fee {
+		priority += c.Amount.Int64()
```

Comment:
```suggestion
                p := c.Amount.Int64()
                if priority == 0 || p < priority {
                  priority = p
                }
```

Reply:
I know you're just taking the minimum here. What does this really buy you? If all fees provided are valid denominations, what is the issue exactly?

Reply:
but the denominations have different prices and settling on the minimum makes it hard to use a cheaper denom alongside the primary denom to boost a tx. the primary denom will take precedence with the minimum

Reply:
I would say this is out of scope of a naive implementation. Its either we spend time designing a new way to do this, which we have pre open for but no reviews, or we document teams how do this and say they have to do it based on their use case. We don't have the man power to design yet another thing right now

Maybe adding a warning godoc could benefit this.

Reply:
But why is there a lot of design needed? It seems like just taking the min solves a lot to potential issues and is a very simple code change

Reply:
I agree with Aaron, what if one denom has exponent = 2, and another one 6? Adding `2 = 0.2` (fee from first denom) to `500 = 0.0005` (fee from the secon denom) doesn't really make sense. So at least we should document / add a comment, that his is a naive implementation.